R/centrality.R

```diff
@@ -1105,13 +1106,13 @@ bonpow.sparse <- function(graph, nodes = V(graph), loops = FALSE,
   vg <- vcount(graph)
 
   ## sparse adjacency matrix
-  d <- as_adj(graph, sparse = TRUE)
+  d <- as_adj(graph, attr = weights, sparse = TRUE)
 
   ## sparse identity matrix
   id <- as(Matrix::Matrix(diag(vg), doDiag = FALSE), "generalMatrix")
 
   ## solve it
-  ev <- Matrix::solve(id - exponent * d, degree(graph, mode = "out"), tol = tol)
+  ev <- Matrix::solve(id - exponent * d, strength(graph, mode = "out"), tol = tol)
```

Comment:
We need to have consistency between the weights used during the adj. mat. conversion and during strength computation. Thus the weights must be passed explicitly here. We need to pay attention and make sure that we handle `NULL`, `NA` or a numeric vector exactly as described in the docs. `strength` might not handle its own `weights` parameter the same way, so we need to check what it does and adapt.